quickwit-core/src/indexing/split.rs

```diff
@@ -127,17 +132,39 @@ impl Split {
             num_parsing_errors: 0,
             storage,
             metastore,
+            timestamp_field,
         })
     }
 
     /// Add document to the index split.
     pub fn add_document(&mut self, doc: Document) -> anyhow::Result<()> {
-        //TODO: handle time range when docMapper is available
+        let mut computed_time_range = None;
+        if let Some(timestamp_field) = self.timestamp_field {
+            let mut split_time_range = self.metadata.time_range.clone().unwrap_or(Range {
+                start: u64::MAX,
+                end: u64::MIN,
+            });
+
+            if let Some(timestamp) = doc
```

Comment:
sounds like we are doing extra stuff here just because we have a tantivy document that is not aware of the timestamp field.
Moreover here we need to know which type we have and... actually it's not a u64. Can you change that ?

Can you also add a todo here so that we find another way to do that later ? I have some idea but not sure if they are good. we need to see that with @fulmicoton

Reply:
basically I would create a struct QuickwitDocument that will be aware of timestamp stuff. This may be overkill though...

Reply:
I was already changing the u64 to i64 because of the merge. 
As to regard to `time_range.clone()`, my reasoning is that since we are using Range<i64> that is cheap to clone.

Reply:
cool :)

Reply:
yes no over engineering